server/routes/routerRequestHandlers.ts

```diff
@@ -105,10 +108,39 @@ const checkNewGoalsFormExistsInSession = async (req: Request, res: Response, nex
   }
 }
 
+/**
+ *  Middleware function that returns a Request handler function to look up the prisoner from prisoner-search, map to a PrisonerSummary, and store in the session
+ */
+const retrievePrisonerSummaryIfNotInSession = (prisonerSearchService: PrisonerSearchService): RequestHandler => {
```

Comment:
This new middleware function is a rework/improvement of what was previously in the now deleted `server/routes/overview/prisonerSummaryRequestHandler.ts`